editor/components/inserter-with-shortcuts/index.js

```diff
@@ -51,13 +51,29 @@ export default compose(
 
 		return {
 			isLocked: !! templateLock,
-			enabledBlockTypes: blockTypes,
+			globalEnabledBlockTypes: blockTypes,
 		};
 	} ),
 	connect(
-		( state, { enabledBlockTypes } ) => ( {
-			items: getFrecentInserterItems( state, enabledBlockTypes, 4 ),
-		} )
+		( state, { globalEnabledBlockTypes } ) => {
+			const insertionPoint = getBlockInsertionPoint( state );
+			const { rootUID } = insertionPoint;
+			const blockListSettings = getBlockListSettings( state, rootUID );
```

Comment:
Maybe we could move this logic to a selector to test it properly:

`getSupportBlocks( state, uid, enabledBlockTypes )`

Reply:
Second that 👍